### PR TITLE
wallet: Select enough XCH to cover the fee when creating an offer

### PR DESCRIPTION
**Pull request by xdustinface**
### Purpose:

Just cleanup / remove some extra logic from `Wallet._generate_unsigned_transaction` which was added in #14812 to get enough XCH amount to also pay the fee when creating the XCH side of an offer.  Just select enough coins during offer creation to also cover the fee instead of selecting extra coins for the fee in `Wallet._generate_unsigned_transaction`.

I think we should either pass in coins into this method or not. If not, we select what we need and if we pass coins in it should be enough to cover everything.

### New Behavior:

No change in behaviour.

### CODE REVIEW

File: chia/wallet/trade_manager.py [hunk 1]
```diff
@@ -507,8 +507,12 @@ async def _create_offer_for_ids(
                         wallet = await self.wallet_state_manager.get_wallet_for_asset_id(asset_id.hex())
                     if not callable(getattr(wallet, "get_coins_to_offer", None)):  # ATTENTION: new wallets
                         raise ValueError(f"Cannot offer coins from wallet id {wallet.id()}")
+                    # For the XCH wallet also include the fee amount to the coins we use to pay this offer
+                    get_coin_amount = abs(amount)
+                    if wallet.type() == WalletType.STANDARD_WALLET:
+                        get_coin_amount += fee
                     coins_to_offer[id] = await wallet.get_coins_to_offer(
-                        asset_id, uint64(abs(amount)), min_coin_amount, max_coin_amount
+                        asset_id, uint64(get_coin_amount), min_coin_amount, max_coin_amount
```
Comment (trepca) [inline chia/wallet/trade_manager.py:515]: `get_coin_amount` sounds like a function, maybe rename to `total_coin_amount_with_fee` or something similar?

Comment (xdustinface) [reply chia/wallet/trade_manager.py]: What about 72d90e4a5411e8e8383e8149d97c7634d9c74315?

File: chia/wallet/trade_manager.py [hunk 2]
```diff
@@ -547,7 +551,11 @@ async def _create_offer_for_ids(
 
             all_transactions: List[TransactionRecord] = []
             fee_left_to_pay: uint64 = fee
-            for id, selected_coins in coins_to_offer.items():
+            # TODO: Create the transaction in place above when we select the coins? The access of the sorted keys here
+            #       makes sure we create the XCH transaction first to make sure we pay fee with the XCH side of the
+            #       offer and don't create an extra fee transaction in other wallets.
```
Comment (Quexington) [inline chia/wallet/trade_manager.py:556]: You're not going to be able to create the transaction above.  Each transaction needs to assert an announcement that contains a hash of all the coins used in the entire offer.  This is why we must select the coins first -> calculate the bundle announcement -> create the transactions.  That part of this comment (assuming I read it correctly) should be deleted.

**Review by Quexington — CHANGES_REQUESTED** [review]

**Review by Quexington — APPROVED** [review]

**Review by trepca — APPROVED** [review]
thanks!

**Pull request merged**